template_functions.go

```diff
@@ -22,8 +22,8 @@ func datacentersFunc(brain *Brain,
 
 		addDependency(used, d)
 
-		if result, ok := brain.datacenters[d.HashCode()]; ok {
-			return result, nil
+		if brain.Remembered(d) {
```

Comment:
This has the potential race condition for the metric to bet forgotten between `.Remembered()` and `.Recall()`, since you give up the lock in between. It may be unlikely with the current code, but it suggests this isn't quite the right API.